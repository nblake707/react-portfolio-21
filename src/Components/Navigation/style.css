.nav-color {
    background-color: #f3f3f3; 
}

.navbar-brand {
    margin-right: .5rem !important;
}

.padding-nav {
    margin-right: 3%;
    margin-left: 2%;
    margin-top: 1%;
}

.bold-brand-name {
    font-weight: 600;
}

.nav-link-border {
    border-right: 1px solid black;
    line-height: 4px;
}

.nav-link-border:last-child {
    border:none;
}
