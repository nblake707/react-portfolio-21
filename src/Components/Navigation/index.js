import React from "react";
import { Link } from 'react-router-dom';
import { Navbar, Nav } from 'react-bootstrap';
import './style.css';

/* <div className="logo">
  <div className="logo-pic"></div>
  <h2>Nateyana Blake</h2>
</div> */
  /* <Link to="/contact">contact</Link>
  <Link to="/resume">resume</Link>
  <Link to="/projects">projects</Link> */

const Navigation = () => {
  return (
      <Navbar className='nav-color padding-nav' expand="lg">
          <Navbar.Brand href="#home" className='bold-brand-name'>Nateyana Blake</Navbar.Brand>
          <Navbar.Text>Web Developer</Navbar.Text>
          <Navbar.Toggle aria-controls="basic-navbar-nav" />
          <Navbar.Collapse id="basic-navbar-nav"> 
            <Nav className='ms-auto'> 
              <Nav.Link href="#home"className='nav-link-border' data-testid='projects'>Projects</Nav.Link>
              <Nav.Link href="#link" className='nav-link-border' data-testid='resume'>Resume</Nav.Link>
              <Nav.Link href="#link" className='nav-link-border' data-testid='contact'>Contact</Nav.Link>
            </Nav>
          </Navbar.Collapse>
      </Navbar>
  );
};

export default Navigation;
