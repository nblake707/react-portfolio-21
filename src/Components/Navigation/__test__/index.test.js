import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Navigation from '../';

afterEach(cleanup);

describe('Navigation component', () => {
    it('renders', () => {
        render(<Navigation />);
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation/>);
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('Navigation links are visible', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Navigation />);
        // Assert
        expect(getByTestId('projects')).toHaveTextContent('Projects');
        expect(getByTestId('resume')).toHaveTextContent('Resume');
        expect(getByTestId('contact')).toHaveTextContent('Contact');
    });
})