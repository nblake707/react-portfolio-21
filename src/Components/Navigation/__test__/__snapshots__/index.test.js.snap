// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="nav-color padding-nav navbar navbar-expand-lg navbar-light"
  >
    <a
      class="bold-brand-name navbar-brand"
      href="#home"
    >
      Nateyana Blake
    </a>
    <span
      class="navbar-text"
    >
      Web Developer
    </span>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="ms-auto navbar-nav"
      >
        <a
          class="nav-link-border nav-link"
          data-rr-ui-event-key="#home"
          data-testid="projects"
          href="#home"
        >
          Projects
        </a>
        <a
          class="nav-link-border nav-link"
          data-rr-ui-event-key="#link"
          data-testid="resume"
          href="#link"
        >
          Resume
        </a>
        <a
          class="nav-link-border nav-link"
          data-rr-ui-event-key="#link"
          data-testid="contact"
          href="#link"
        >
          Contact
        </a>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
